src/main/java/com/internship/socialnetwork/service/impl/FriendshipServiceImpl.java

```diff
@@ -0,0 +1,90 @@
+
+import java.util.List;
+import java.util.Optional;
+
+import static com.internship.socialnetwork.dto.FriendshipDto.convertToDto;
+
+import static com.internship.socialnetwork.model.FriendshipStatus.ACCEPTED;
+import static com.internship.socialnetwork.model.FriendshipStatus.PENDING;
+
+@Service
+@RequiredArgsConstructor
+public class FriendshipServiceImpl implements FriendshipService {
+
+    private static final String FRIENDSHIP_EXISTS_MSG  = "Friendship with ids: %s , %s already exists!";
+
+    private static final String FRIENDSHIP_DOES_NOT_EXIST_MSG  = "Friendship with ids: %s , %s does not exist!";
+
+    private final FriendshipRepository friendshipRepository;
+
+    private final UserService userService;
+
+    @Override
+    public List<FriendshipDto> getFriends(Long userId) {
+        return friendshipRepository.findAllFriends(userId)
+                .stream()
+                .map(FriendshipDto::convertToDto)
+                .toList();
+    }
+
+    @Override
+    public List<FriendshipDto> getFriendRequests(Long userId) {
+        return friendshipRepository.findAllFriendRequests(userId)
+                .stream().map(FriendshipDto::convertToDto)
+                .toList();
+    }
+
+    @Override
+    public List<FriendshipDto> getSentFriendRequests(Long userId) {
+        return friendshipRepository.findAllSentFriendRequests(userId)
+                .stream()
+                .map(FriendshipDto::convertToDto)
+                .toList();
+    }
+
+    @Override
+    public FriendshipDto createFriendRequest(Long userId, Long friendId) {
+        findFriendship(userId, friendId)
+                .ifPresent(friendship -> {
+                    throw new BadRequestException(String.format(FRIENDSHIP_EXISTS_MSG, userId, friendId));
+                });
+
+        return convertToDto(friendshipRepository.save(
+                Friendship.builder()
+                        .user(userService.getUserModel(userId))
+                        .friend(userService.getUserModel(friendId))
+                        .friendshipStatus(PENDING)
+                        .build()));
+    }
+
+    @Override
+    public void deleteFriendship(Long userId, Long friendId) {
+        friendshipRepository.delete(findFriendship(userId, friendId)
+                .orElseThrow( () -> new BadRequestException(String.format(FRIENDSHIP_DOES_NOT_EXIST_MSG, userId, friendId))));
+    }
+
+    @Override
+    public FriendshipDto acceptFriendRequest(Long userId, Long friendId) {
+        Friendship friendship = findFriendship(userId, friendId)
+                .orElseThrow( () -> new BadRequestException(String.format(FRIENDSHIP_DOES_NOT_EXIST_MSG, userId, friendId)));
+        friendship.setFriendshipStatus(ACCEPTED);
+        return convertToDto(friendshipRepository.save(friendship));
+    }
+
+    private Optional<Friendship> findFriendship(Long userId, Long friendId) {
+        return friendshipRepository.findFriendship(userId, friendId);
```

Comment:
Pls add here orElseThrow instead of doing it multiple times above. That means you will not return Optional